crates/miden-objects/src/note/mod.rs

```diff
@@ -173,6 +176,29 @@ impl AsRef<NoteRecipient> for Note {
     }
 }
 
+// TO INPUT NOTE COMMITMENTS
+// ================================================================================================
+
+impl ToInputNoteCommitments for Note {
```

Comment:
I think this works but it may be better to add a dedicated constructor for unauthenticated notes. The main reason is that it may not be immediately obvious that if we create `InputNotes` by passing a vector of notes into, these notes would be treated at unauthenticated. So, at the very least we should document this clearly here, and also in the doc comments for `InputNotes`.

But again, I think just having `InputNotes::from_unauthenticated_notes()` would make this clear by construction.

Reply:
Should I add both the new constructor and extended docs for this trait implementation, or just having the constructor should be sufficient?